components/mediation/data-publishers/org.wso2.micro.integrator.observability/src/main/java/org/wso2/micro/integrator/observability/handler/metrichandler/prometheus/reporter/PrometheusMetricCreatorUtils.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.micro.integrator.observability.handler.metrichandler.prometheus.reporter;
+
+import org.apache.synapse.SynapseConstants;
+import org.wso2.micro.integrator.observability.handler.util.MetricConstants;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Class for creating Prometheus Metrics from the data scraped in the Metric Handler.
+ */
+public class PrometheusMetricCreatorUtils {
+    private static PrometheusReporter prometheusReporter = new PrometheusReporter();
+
+    /**
+     * Create the proxy services related metrics.
+     */
+    public static void createProxyServiceMetric() {
+        Map<String, String[]> map = new HashMap<>();
+        String[] labels = {MetricConstants.SERVICE_NAME, MetricConstants.SERVICE_TYPE};
+        map.put(MetricConstants.PROXY_REQUEST_COUNT_TOTAL, labels);
+        map.put(MetricConstants.PROXY_LATENCY_SECONDS, labels);
+
+         prometheusReporter.createMetric(SynapseConstants.PROXY_SERVICE_TYPE, MetricConstants.COUNTER,
```

Comment:
PrometheusReporter calls PrometheusMetricCreatorUtils and then again new instance of PrometheusReporter is created inside. This is wrong. Either move full creation logic to Utils class or move label population logic to PrometheusReporter class

Reply:
Moved the  label population logic to PrometheusReporter class